From add1fe8e7743ddcbc74865f13910a10cbb6253ab Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sun, 10 Aug 2014 18:21:09 -0400
Subject: [iommu] Change iommu driver to call io_page_fault trace event

Message-id: <20140810182109.24055.30558.stgit@gir.stowe>
Patchwork-id: 87588
O-Subject: [RHEL7.1 PATCH 17/19] iommu: Change iommu driver to call io_page_fault trace event
Bugzilla: 1127467
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1127467
Upstream Status: 56fa484969c367e3ae43a012a7b99f75bb4f3bdb

commit 56fa484969c367e3ae43a012a7b99f75bb4f3bdb
Author: Shuah Khan <shuah.kh@samsung.com>
Date:   Tue Sep 24 15:21:20 2013 -0600

    iommu: Change iommu driver to call io_page_fault trace event

    Change iommu driver call io_page_fault trace event. This iommu_error class
    event can be enabled to trigger when an iommu error occurs. Trace informatio
    includes driver name, device name, iova, and flags.

    Testing:
    Added trace calls to iommu_prepare_identity_map() for testing some of the
    conditions that are hard to trigger. Here is the trace from the testing:

           swapper/0-1     [003] ....     2.003774: io_page_fault: IOMMU:pci 000
           swapper/0-1     [003] ....     2.004098: io_page_fault: IOMMU:pci 000
           swapper/0-1     [003] ....     2.004115: io_page_fault: IOMMU:pci 000
           swapper/0-1     [003] ....     2.004129: io_page_fault: IOMMU:pci 000

    Signed-off-by: Shuah Khan <shuah.kh@samsung.com>
    Signed-off-by: Joerg Roedel <joro@8bytes.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/include/linux/iommu.h b/include/linux/iommu.h
index 3aeb730..5fcb0d3 100644
--- a/include/linux/iommu.h
+++ b/include/linux/iommu.h
@@ -22,6 +22,7 @@
 #include <linux/errno.h>
 #include <linux/err.h>
 #include <linux/types.h>
+#include <trace/events/iommu.h>
 
 #define IOMMU_READ (1)
 #define IOMMU_WRITE (2)
@@ -211,6 +212,7 @@ static inline int report_iommu_fault(struct iommu_domain *domain,
   ret = domain->handler(domain, dev, iova, flags,
       domain->handler_token);
 
+ trace_io_page_fault(dev, iova, flags);
  return ret;
 }
 
-- 
1.7.1